### ISSUE TITLE
How to bump versions on refactor?

### ISSUE THREAD

**Issue by rarkins** [semantic-release/semantic-release#472]
With the current v8 of this repo, what is the way to bump up the patch version if I use the `refactor:` prefix? Does it still require a plugin or can be configured immediately in `package.json`?

i.e. I want every refactor to cause a patch release

**Comment by pvdlg** [comment]
You can configure that as explain in [commit-analyzer#release-rules](https://github.com/semantic-release/commit-analyzer#release-rules).

**Comment by gr2m** [comment]
out of curiosity, why would you want to release a new version if you just refactored without any behaviour changes? If it’s a speed improvement worth releasing, I’d use a fix prefix

**Comment by rarkins** [comment]
@gr2m I think that the default decision to leave refactoring out of releases goes against the higher level semantic-release philosophy of "remove the immediate connection between human emotions and version numbers", probably because we all don't want it to "look like another bug fix". Instead, I think semver's biggest value to the downstream user is to indicate risk, not characteristic, as represented here:

| type  | characteristic | breakage               | risk   |
|-------|----------------|------------------------|--------|
| major | breaking       | definitely broke       | high   |
| minor | feature        | maybe broke            | medium |
| patch | fix            | hopefully didn't break | low    |

But refactoring is usually a risk too, sometimes even a medium risk. If I've done a *big* refactor then I even prefer to even release it as a minor version bump to indicate medium risk. Hence:

major = backwards-incompatible change
minor = code with risk of breakage, you may want to check it
patch = should only ever improve stability

tldr; I prefer an approach of breaking/medium risk/low risk instead of breaking/feature/fix, because that's what the end user is ultimately trying to judge anyway when they look at versions or read release notes.
